Hide channel name in topic header bars if it is already displayed in navbar.

pratik-pc opened the pull request:
Earlier, channel was displayed in recipient row for message groups in all views.

This PR omits channel name from recipient row in message groups if it is already displayed in navbar.

Fixes: zulip#34480.

<!-- Describe your pull request here.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

|**Before:**|**After:**|
|-|-|
|![Screenshot from 2025-05-23 20-44-54](https://github.com/user-attachments/assets/0bbc1f6f-5ff4-4e01-9a1d-747baf51fde6)|![Screenshot from 2025-06-06 15-51-52](https://github.com/user-attachments/assets/63e3e25b-1b1f-40ca-af95-2d407ee4cdc2)|


|**Before:**|**After:**|
|-|-|
|<img width="828" height="336" alt="image" src="https://github.com/user-attachments/assets/bc05146d-d54e-42ef-be40-37b3d6df6ee1" />|<img width="534" height="321" alt="image" src="https://github.com/user-attachments/assets/833d21f6-18ee-42a0-9796-2675b1d246d3" />|
|<img width="828" height="336" alt="image" src="https://github.com/user-attachments/assets/3ce4f8e1-a9df-4f54-a247-a8e9f8fb85c9" />|<img width="493" height="220" alt="image" src="https://github.com/user-attachments/assets/6d886483-3f76-4131-9dfc-c6660477e351" />|
|<img width="1589" height="600" alt="image" src="https://github.com/user-attachments/assets/ca26a73c-d757-4c14-8261-de4f3bb35737" />|<img width="1574" height="515" alt="image" src="https://github.com/user-attachments/assets/9c79cc03-3b1b-4687-b410-62856802c848" />|


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

pratik-pc committed:
recipient_row: Omit channel name if it is already displayed in navbar.

Earlier, channel was displayed in recipient row for message groups
in all views.

This commit omits channel name from recipient row in message groups
if it is already displayed in navbar.

Fixes: zulip#34480.

alya commented:
The channel privacy symbol should be hidden as well, I think. I was thinking of that as part of the channel name.

pratik-pc commented:
> The channel privacy symbol should be hidden as well, I think. I was thinking of that as part of the channel name.

- Should `(archived)` stay alongside topic name?
- if yes, should clicking on it narrow to stream feed?

alya commented:
No, that info is available in the navbar, and we should drop it from the header bars.

opmkumar reviewed (commented):
This PR slipped through my mind. Sorry for the delayed review, @pratik-pc.
Apart from these couple of comments, this PR looks good to me.4

Let me know if you have any questions about my comments below!

pratik-pc commented on web/templates/recipient_row.hbs:
```diff
@@ -1,7 +1,8 @@
 {{#if is_stream}}
 <div class="message_header message_header_stream right_part" data-stream-id="{{stream_id}}">
-    <div class="message-header-contents" style="background: {{recipient_bar_color}};">
+    <div class="message-header-contents" style="background: {{recipient_bar_color}}; {{#if channel_view}}padding-left: 10px;{{/if}}">
```
Changed to accommodate this

pratik-pc commented:
@zulipbot remove "buddy review"
@zulipbot add "mentor review"

@ligmitz up for your review

@opmkumar feel free to check if it makes sense now.

pratik-pc commented:
@alya passed mentor review

alya commented:
Works for me in manual testing!

@amanagr Are you up for reviewing this one? The next step is to test-deploy on CZO to text the UX, so the code doesn't necessarily need to be perfect for now.

timabbott commented on web/e2e-tests/lib/common.ts:
```diff
@@ -505,16 +506,33 @@ export async function get_rendered_messages(
+                key =
+                    (await page.title()) === "Combined feed - Zulip Dev - Zulip"
+                        ? `${stream_name} > ${topic_name}`
+                        : topic_name;
```
This code is really hard to read. I suggest using fewer `?` comprehensions and using normal if/else statements, especially for situations like the topic_name being set which you check multiple times.

timabbott commented on web/src/message_list_view.ts:
```diff
@@ -1655,7 +1658,9 @@ export class MessageListView {
 
-        const $rendered_recipient_row = $(render_recipient_row(group));
+        const current_filter = narrow_state.filter();
+        const channel_view = current_filter?.can_show_next_unread_topic_conversation_button();
```
This might happen to do a valid calculation, but it is super confusing. 

Why aren't you calling `is_channel_view`?

timabbott commented:
Did a quick review; I should also note the commit message has the same typo a few of your other PRs have had with the "Fixes" tag, where it won't actually close the target issue. You need a space. Can you audit all your commits for this bug?

pratik-pc committed:
css: Remove extra space above & below top message loading indicator.

We define a .top-messages-logo-wrapper which takes up enough space
so that the navbar bottom padding(10px) and sticky message header top
margin (10px) together make up 2.7143em(size of spinner). We place both
the logo and spinner at the center.

amanagr commented:
Thanks, looks good to me now! Tested manually as well.

amanagr commented:
This is waiting on design discussion in [#design > channel name in recipient headers](https://chat.zulip.org/#narrow/channel/101-design/topic/channel.20name.20in.20recipient.20headers/with/2213018) which has grown stale.
